packages/compiler/src/core/helpers/syntax-utils.ts

```diff
@@ -0,0 +1,45 @@
+import { isIdentifierContinue, isIdentifierStart, utf16CodeUnits } from "../charcode.js";
+import { Keywords } from "../scanner.js";
+
+/**
+ * Print a string as a TypeSpec identifier. If the string is a valid identifier, return it as is otherwise wrap it into backticks.
+ * @param sv Identifier string value.
+ * @returns Identifier string as it would be represented in a TypeSpec file.
+ *
+ * @example
+ * ```ts
+ * printIdentifier("foo") // foo
+ * printIdentifier("foo bar") // `foo bar`
+ * ```
+ */
+export function printIdentifier(sv: string) {
```

Comment:
open to a different name knowing this will be accessible as 
```
import {printIdentifier} from "@typespec/compiler";
```

Reply:
`printIndentifierSyntax` could be more explicit

Reply:
I'm fine with the current name - matches the description/seems reasonable.